doc/elasticdl_on_sqlflow.md

```diff
@@ -0,0 +1,141 @@
+# _Design:_ ElasticDL on sqlflow
+
+## Overview
+
+This is a design doc on integration with [ElasticDL](https://github.com/wangkuiyi/elasticdl).
+
+### User Interface
+
+#### Training Job Submission
+
+```sql
+SELECT 
+    c1, c2, c3, c4, c5 as class
+FROM training_data
+TRAIN ElasticDLKerasClassifier
+WITH
+  optimizer = "optimizer"
+  loss = "loss"
+  eval_metrics = "eval_metrics_fn"
+  num_classes = 10
+COLUMN
+  c1,
+  NUMERIC(c2, 10),
+  BUCKET(c3, [0, 10, 100]),
+  c4
+LABEL class
+INTO trained_elasticdl_keras_classifier;
+```
+
+#### Prediction Job Submission
+
+```sql
+SELECT 
+    c1, c2, c3, c4
+FROM prediction_data
+PREDICT prediction_results_table
+WITH
+  num_classes = 10
+USING trained_elasticdl_keras_classifier;
+```
+
+#### Run-time Configurations
+
+Users can provide run-time configurations to ElasticDL job via additional parameters with prefix "runtime" within `WITH` clause, for example:
+
+```sql
+SELECT
+    c1, c2, c3, c4, c5 as class
+FROM training_data
+TRAIN ElasticDLKerasClassifier
+WITH
+  optimizer = "optimizer"
+  loss = "loss"
+  eval_metrics = "eval_metrics_fn"
+  num_classes = 10
+  runtime.num_epochs = 2
+  runtime.master_resource_request = "cpu=400m,memory=1024Mi"
+  runtime.master_resource_limit = "cpu=400m,memory=1024Mi"
+  runtime.worker_resource_request = "cpu=400m,memory=2048Mi"
+  runtime.worker_resource_limit = "cpu=1,memory=3072Mi"
```

Comment:
It is too complicated for SQL users to specify worker/master resources. Each predefined model can have an optimal resource config to be used directly.

Reply:
Yes, another approach is to pre-define some resource types like "small", "standard", "large". But from API side, we still provide a way for advanced users to config their tasks, right? Not sure how ALPS handles this.

Reply:
Just checked with ALPS folks. They handle this types of configurations in WITH clause as well.